src/datajudge/constraints/interval.py

```diff
@@ -0,0 +1,131 @@
+import abc
+from typing import Any, List, Optional, Tuple
+
+import sqlalchemy as sa
+
+from .. import db_access
+from ..db_access import DataReference
+from .base import Constraint, OptionalSelections
+
+
+class IntervalConstraint(Constraint, abc.ABC):
```

Comment:
Any reason to inherit explicitly from `abc.ABC`? Because `Constraint` already does.
```suggestion
class IntervalConstraint(Constraint):
```